src/ninshiki_cpp/detector/lbp_detector.cpp

```diff
@@ -49,22 +49,22 @@ bool LBPDetector::loadClassifier(std::string config_path) {
     return true;
 }
 
-void LBPDetector::detection(cv::Mat input)
+void LBPDetector::detection(const cv::Mat & input)
 {
-    img_width = static_cast<double>(input.cols);
-    img_height = static_cast<double>(input.rows);
-
-    std::vector<cv::Rect> rects;
     if (!classifier_loaded)
         return;
 
-    cv::Mat gray = input.clone();
+    cv::Mat gray = input;
     cv::cvtColor(input, gray, cv::COLOR_BGR2GRAY);
```

Comment:
better to keep `gray` unitialized, since it will be rewritten by `cv::cvtColor()` function.